### PR TITLE
Add testing to cmake

### PR DESCRIPTION
**Pull request by wow2006**

### CODE REVIEW

File: tests/auto/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+target_compile_definitions(
+  test-app
+  PRIVATE
+  QT_DEPRECATED_WARNINGS
+)
+
+add_test(
+  test-app
+  test-app
```
Comment (mitchcurtis) [inline tests/auto/CMakeLists.txt]: Does this need to be here twice?

File: app/CMakeLists.txt [hunk 2]
```diff
@@ -7,6 +7,7 @@ if(WIN32)
 else()
   add_executable(
     app
+    ""
```
Comment (mitchcurtis) [inline app/CMakeLists.txt]: Is this empty string supposed to be here?

**Comment by mitchcurtis** [comment]
Thanks! Looks mostly good, just a few questions.

File: app/CMakeLists.txt (hunk 2 above)
Comment (wow2006) [reply app/CMakeLists.txt]: I will fix it. I add it coz old cmake versions don't support `add_executable` with the target only.

**Comment by wow2006** [comment]
I just remove the empty string from `add_executable`.

**Comment by mitchcurtis** [comment]
What about the other comment? "test-app" is duplicated twice - and that happens for each test. I dunno if it matters but it looks fishy. :)

**Comment by mitchcurtis** [comment]
OK, thanks! :)

**Pull request merged**